wcfsetup/install/files/lib/acp/form/PackageUpdateServerAddForm.class.php

```diff
@@ -63,13 +65,45 @@ public function readFormParameters() {
 	public function validate() {
 		parent::validate();
 		
+		$this->validateServerURL();
+	}
+	
+	/**
+	 * Validates the server URL.
+	 */
+	protected function validateServerURL() {
 		if (empty($this->serverURL)) {
 			throw new UserInputException('serverURL');
 		}
 		
 		if (!PackageUpdateServer::isValidServerURL($this->serverURL)) {
 			throw new UserInputException('serverURL', 'invalid');
 		}
+		
+		if (preg_match('/^.*\.woltlab.com$/', Url::parse($this->serverURL)['host'])) {
```

Comment:
Maybe use `StringUtil::endsWith()` instead, that's a bit more verbose in terms of documenting the intention of the regex (also, your regex is not case-insensitive).